src/Batch/BatchApiV2.php

```diff
@@ -0,0 +1,161 @@
+
+use GuzzleHttp\ClientInterface;
+use JetBrains\PhpStorm\ArrayShape;
+use JetBrains\PhpStorm\ExpectedValues;
+use Psr\Log\LoggerInterface;
+use Smartling\AuthApi\AuthApiInterface;
+use Smartling\BaseApiAbstract;
+use Smartling\Batch\Params\ListBatchesParameters;
+use Smartling\Exceptions\SmartlingApiException;
+
+class BatchApiV2 extends BaseApiAbstract
+{
+    public const ENDPOINT_URL = 'https://api.smartling.com/jobs-batches-api/v2/projects';
+    private const GET_BATCH_STATUS_ARRAY_SHAPE = [
+        'authorized' => 'bool',
+        'files' => 'array',
+        'generalErrors' => 'string',
+        'projectId' => 'string',
+        'status' => 'string',
+        'translationJobUid' => 'string',
+        'updatedDate' => 'string',
+    ];
+
+    public function __construct(
+        AuthApiInterface $authProvider,
+        string $projectId,
+        LoggerInterface $logger = null,
+        ClientInterface $client = null
+    )
+    {
+        if ($client === null) {
+            $client = self::initializeHttpClient(self::ENDPOINT_URL);
+        }
+        parent::__construct($projectId, $client, $logger, self::ENDPOINT_URL);
+        $this->setAuth($authProvider);
+    }
+
+    /**
+     * @throws SmartlingApiException
+     */
+    #[ArrayShape(['batchUid' => 'string'])]
+    public function createBatch(bool $authorize,
+        string $translationJobUid,
+        array $fileUris,
+        array $localeWorkflows = []
+    ): array
+    {
+        if (count($fileUris) === 0) {
+            throw new \UnexpectedValueException('FileUris cannot be empty.');
+        }
+        $parameters = [
+            'authorize' => $authorize,
+            'translationJobUid' => $translationJobUid,
+            'fileUris' => $fileUris,
+        ];
+        if (count($localeWorkflows) !== 0) {
+            $parameters['localeWorkflows'] = $localeWorkflows;
+        }
+        return $this->sendRequest(
+            'batches',
+            $this->getDefaultRequestData('json', $parameters),
+            self::HTTP_METHOD_POST,
+        );
+    }
+
+    /**
+     * @throws SmartlingApiException
+     */
+    public function listBatches(ListBatchesParameters $parameters): array
+    {
+        return $this->sendRequest(
+            "batches",
+            $this->getDefaultRequestData('query', $parameters->exportToArray()),
+            self::HTTP_METHOD_GET,
+        );
+    }
+
+    /**
+     * @throws SmartlingApiException
+     */
+    #[ArrayShape(self::GET_BATCH_STATUS_ARRAY_SHAPE)]
+    public function getBatchStatus(string $batchUid): array
+    {
+        $this->assertBatchUid($batchUid);
+        return $this->sendRequest(
+            "batches/$batchUid",
+            $this->getDefaultRequestData('query', []),
+            self::HTTP_METHOD_GET,
+        );
+    }
+
+    /**
+     * @throws SmartlingApiException
+     */
+    public function processBatchAction(
+        string $batchUid,
+        #[ExpectedValues(['CANCEL_FILE', 'REGISTER_FILE'])]
```

Comment:
The same. It's only for autocompletion, it's ridiculous. You can put whatever you want in `$action` - nothing happens.